posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,377 @@
+        responseObserver.onNext(reply);
+        responseObserver.onCompleted();
+    }
+}
+----
+
+For this example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. No additional libraries need to be provided with the application, and once it's started the helloworld greeter service will be accessible on the server's HTTP endpoints.
+
+For a client example, a basic Servlet using gRPC can be defined using the `grpcClient-1.0` feature with:
+
+[source, java]
+----
+package com.ibm.ws.grpc;
+
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+...
+@WebServlet(name = "grpcClient", urlPatterns = { "/grpcClient" }, loadOnStartup = 1)
+public class GrpcClientServlet extends HttpServlet {
+
+        ManagedChannel channel;
+        private GreeterGrpc.GreeterBlockingStub greetingService;
+
+        private void startService(String address, int port) 
+        {
+            channel = ManagedChannelBuilder.forAddress(address , port).usePlaintext().build();
+            greetingService = GreeterGrpc.newBlockingStub(channel);
+        }
+
+        private void stopService() 
+        {
+            channel.shutdownNow();
+        }
+
+        @Override
+        protected void doGet(HttpServletRequest reqest, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+            // set user, address, port params
+        }
+
+        @Override
+        protected void doPost(HttpServletRequest request, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+        // grab user, address, port params
+        startService(address, port);
+        HelloRequest person = HelloRequest.newBuilder().setName(user).build();
+        HelloReply greeting = greetingService.sayHello(person);
+
+        // send the greeting in a response
+        stopService();
+        }	
+    }
+}
+----
+
+As with the service example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. All required gRPC client libraries are provided by the `grpcClient-1.0` feature.
+
+[#jndi]
+=== Support for Custom JNDI Names for Enterprise Beans
+
+[#support]
+==== What is Support for Custom JNDI Names for Enterprise Beans?
+
+Support for Custom JNDI Names for Enterprise Beans (EJBs) is an enhancement to all existing enterprise beans features that allows an application to configure and use custom JDNI names for looking up and/or injecting enterprise beans, and to also use legacy default JNDI names from WebSphere traditional.
+
```

Comment:
Let's change: "and to also use legacy default JNDI names from WebSphere traditional."
to "or use legacy default JNDI names instead of the specification defined JNDI names"

This eliminates the tWAS reference and indicates you may now use the 1) spec defined names, 2) the custom names or 3) the legacy defaults.  I think indicating "legacy" is important to differentiate these from the spec defined defaults.

Reply:
Done